lsp/client-extension/syntaxes/nickel.tmLanguage.json

```diff
@@ -0,0 +1,137 @@
+{
+  "$schema": "https://raw.githubusercontent.com/martinring/tmlanguage/master/tmlanguage.json",
+  "name": "Nickel",
+  "patterns": [
+    {
+      "include": "#keywords"
+    },
+    {
+      "include": "#strings"
+    },
+    {
+      "include": "#comments"
+    },
+    {
+      "include": "#operators"
+    },
+    {
+      "include": "#numbers"
+    },
+    {
+      "include": "#bools"
+    },
+    {
+      "include": "#storage"
+    },
+    {
+      "include": "#types"
+    },
+    {
+      "include": "#strings_interpolation"
+    }
+  ],
+  "repository": {
+    "storage": {
+      "patterns": [
+        {
+          "name": "storage.modifier",
+          "match": "\\b(let|in|forall|import|default|force|optional|priority|doc|rec)\\b"
+        },
+        {
+          "name": "storage.type",
+          "match": "\\b(fun)\\b"
+        }
+      ]
+    },
+    "strings_interpolation": {
+      "name": "string.interpolated",
+      "begin": "m%\"$",
```

Comment:
As a note, this won't work with variable length string delimiters: `m%%\" some string "%%`, but honestly we can add this in a second step. To do that properly we need to share some state between `begin`, `end`, and `patterns`, which would be set by `begin` (and is the number of `%` in the opening delimiter).

Reply:
Yeah I'm not too familiar with the `tmLanguage` spec. I added a fix that should at least handle the start and end of variable length string delimiters.